internal/querynodev2/segments/segment_loader.go

```diff
@@ -70,10 +70,42 @@ type Loader interface {
 	LoadBloomFilterSet(ctx context.Context, collectionID int64, version int64, infos ...*querypb.SegmentLoadInfo) ([]*pkoracle.BloomFilterSet, error)
 }
 
+func NewLoader(
+	manager CollectionManager,
+	cm storage.ChunkManager,
+) *segmentLoader {
+	cpuNum := runtime.GOMAXPROCS(0)
+	ioPoolSize := cpuNum * 8
+	// make sure small machines could load faster
+	if ioPoolSize < 32 {
+		ioPoolSize = 32
+	}
+	// limit the number of concurrency
+	if ioPoolSize > 256 {
+		ioPoolSize = 256
+	}
+
+	if configPoolSize := paramtable.Get().QueryNodeCfg.IoPoolSize.GetAsInt(); configPoolSize > 0 {
```

Comment:
does `at most 256` rules apply to config value?

Reply:
And `at least 32` rule?

Reply:
No